sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2359,29 +2359,30 @@ pi_result piEventGetProfilingInfo(pi_event Event, pi_profiling_info ParamName,
                                   size_t *ParamValueSizeRet) {
 
   assert(Event);
-  uint64_t ZeTimerResolution =
-      Event->Queue->Context->Device->ZeDeviceProperties.timerResolution;
 
   ReturnHelper ReturnValue(ParamValueSize, ParamValue, ParamValueSizeRet);
+
   switch (ParamName) {
   case PI_PROFILING_INFO_COMMAND_START: {
-    uint64_t ContextStart;
-    ZE_CALL(zeEventGetTimestamp(
-        Event->ZeEvent, ZE_EVENT_TIMESTAMP_CONTEXT_START, &ContextStart));
-    ContextStart *= ZeTimerResolution;
-    return ReturnValue(uint64_t{ContextStart});
+    ze_kernel_timestamp_result_t* tsResult = nullptr;
+    ZE_CALL(zeCommandListAppendQueryKernelTimestamps(
```

Comment:
The spec says https://spec.oneapi.com/level-zero/latest/core/api.html?highlight=zecommandlistappendquerykerneltimestamps#_CPPv440zeCommandListAppendQueryKernelTimestamps24ze_command_list_handle_t8uint32_tP17ze_event_handle_tPvPK6size_t17ze_event_handle_t8uint32_tP17ze_event_handle_t:

> Appends a query of an events’ timestamp value(s) into a command list.

I don't see how this is going to achieve what's needed. It looks like these timestamp query should be appended around the command being profiled. Could you explain how this is working?